x/epochstorage/keeper/stake_storage.go

```diff
@@ -236,26 +241,23 @@ func (k Keeper) AppendStakeEntryCurrent(ctx sdk.Context, chainID string, stakeEn
 		}
 	}
 	stakeStorage.StakeEntries = entries
-	k.SetStakeStorageCurrent(ctx, chainID, stakeStorage)
+	k.SetStakeStorage(ctx, stakeStorage)
 }
 
-func (k Keeper) ModifyStakeEntryCurrent(ctx sdk.Context, chainID string, stakeEntry types.StakeEntry, removeIndex uint64) {
-	// this stake storage entries are sorted by stake amount
-	stakeStorage, found := k.GetStakeStorageCurrent(ctx, chainID)
-	if !found {
-		// should not happen since caller is expected to validate chainID first;
-		// do nothing and return to avoid panic.
-		utils.LavaFormatError("critical: ModifyStakeEntryCurrent with unknown chain", legacyerrors.ErrNotFound,
-			utils.LogAttr("chainID", chainID),
-			utils.LogAttr("stakeAddr", stakeEntry.Address),
-		)
-		return
-	}
+func (k Keeper) AppendStakeEntryCurrent(ctx sdk.Context, chainID string, stakeEntry types.StakeEntry) {
+	k.AppendStakeEntryFromStorage(ctx, k.stakeStorageKeyCurrent(chainID), stakeEntry, false)
+}
+
+func (k Keeper) ModifyStakeEntryCurrentFromStorage(ctx sdk.Context, stakeStorage types.StakeStorage, stakeEntry types.StakeEntry) {
```

Comment:
this can be a method of the stakestorage and not the keeper

Reply:
I can't do that because I use the keeper